rspace/src/test/scala/coop/rchain/rspace/IStoreTests.scala

```diff
@@ -0,0 +1,100 @@
+package coop.rchain.rspace
+
+import coop.rchain.rspace.examples.StringExamples._
+import coop.rchain.rspace.examples.StringExamples.implicits._
+import coop.rchain.rspace.extended._
+import coop.rchain.rspace.internal._
+import coop.rchain.rspace.test._
+import org.scalacheck.Gen
+import org.scalactic.anyvals.PosInt
+import org.scalatest._
+import org.scalatest.prop.GeneratorDrivenPropertyChecks
+
+trait IStoreTests
+    extends StorageTestsBase[String, Pattern, String, StringsCaptor]
+    with GeneratorDrivenPropertyChecks {
+
+  implicit override val generatorDrivenConfig: PropertyCheckConfiguration =
+    PropertyCheckConfiguration(minSuccessful = PosInt(1000))
+
+  "putDatum" should "put datum in a new channel" in withTestStore { store =>
+    forAll("channel", "datum") { (channel: String, datumValue: String) =>
+      val key   = List(channel)
+      val datum = Datum(datumValue, persist = false)
+
+      store.withTxn(store.createTxnWrite()) { txn =>
+        store.putDatum(txn, key, datum)
+        store.getData(txn, key) should contain theSameElementsAs (Seq(datum))
+      }
+      store.clear()
```

Comment:
Is this necessary?

Reply:
Unfortunately yes, because `store` is shared between tests run for different generated values. I could move `forAll` above `withTestStore`, but then I'd have to put generated values inside test names, e.g. `"putDatum" should s"put datum $datumValue in a new channel $channel"` so there are no conflicts, but this would pollute the console

Reply:
Oh right, yep - I see now.  Thanks for the clarification.  No need to change anything.